pkg/jobs/jobs_test.go

```diff
@@ -116,33 +120,108 @@ spec:
+                       name:        "test-job",
+                       namespace:   testNamespace,
+                       status:      "Complete",
+                       delay:       1 * time.Second,
+                       expectError: false,
+                       wait:        "2s",
+               },
+               {
+                       test:        "timeout waiting job complete",
+                       name:        "test-job",
+                       namespace:   testNamespace,
+                       status:      "Complete",
+                       delay:       10 * time.Second,
+                       expectError: true,
+                       wait:        "2s",
+               },
+               {
+                       test:        "wait failed job",
+                       name:        "pod-running",
+                       namespace:   testNamespace,
+                       status:      "Failed",
+                       delay:       1 * time.Second,
+                       expectError: true,
+                       wait:        "2s",
+               },
+       }
+       for _, tc := range testCases {
+               t.Run(tc.test, func(t *testing.T) {
+                       // TODO Figure out the rest.Config
+                       client := fake.NewSimpleClientset()
+                       watcher := watch.NewFake()
+                       client.PrependWatchReactor("jobs", k8stest.DefaultWatchReactor(watcher, nil))
+                       fixture := New(client, metav1.ListOptions{}, nil)
+                       go func(tc TestCase) {
+                               time.Sleep(tc.delay)
+                               watcher.Modify(testutils.NewJobWithStatus(tc.name, tc.namespace, tc.status))
+                       }(tc)
 
-	if err != nil {
-		t.Errorf("encountered an error: %v", err)
-		return
-	}
-	if result.Name != options.Name || result.Namespace != options.Namespace {
-		t.Errorf("incorrect instance was returned")
-		return
-	}
-	jobs, _ := fixture.List(testNamespace)
-	if len(jobs) != 2 {
-		t.Errorf("expecting 2 jobs in namespace, listing returned %v", len(jobs))
-		return
-	}
+                       result, err := fixture.Create(JobOptions{
+                               Name:          tc.name,
+                               Namespace:     tc.namespace,
+                               Image:         "busybox",
+                               Command:       []string{"sh", "-c", "sleep 300"},
+                               RestartPolicy: k8sTypes.RestartPolicyNever,
+                               Wait:          tc.wait,
+                       })
+
+                       if !tc.expectError && err != nil {
+                               t.Errorf("unexpected error: %v", err)
+                               return
+                       }
+                       if tc.expectError && err == nil {
+                               t.Errorf("Expected an error but none returned")
```

Comment:
Minor nitpicks for any subsequent update:

* static string can use t.Error("...")
* the error text should start with lower-case, i.e. expected vs. Expected as this would trigger a linting error once enabled